backend/msgpack/queue.go

```diff
@@ -120,21 +125,43 @@ func (q *queue) drain() {
 				xlog.NewLogField("instance", q.instance.Endpoint()),
 				xlog.NewLogErrField(err),
 			).Error("write data error")
-			q.metrics.connWriteErrors.Inc(1)
 		}
 		buf.Close()
 	}
 	q.conn.Close()
 }
 
+func (q *queue) reportQueueSize(reportInterval time.Duration) {
+	ticker := time.NewTicker(reportInterval)
+	for range ticker.C {
+		q.RLock()
```

Comment:
I'd actually add a `doneCh = make(chan struct{})` to the `queue`, close it on queue close, and select on the `doneCh` instead of repeatedly locking and unlocking the queue.